tests/test_tcp_streamer_block.py

```
@@ -42,5 +47,77 @@ def test_other_configurations(self):
             mock_socket.return_value.__enter__.return_value.\
                 bind.assert_called_once_with(('1.2.3.4', 12345))
 
-    # TODO: test no data
-    # TODO: test data after no data
\ No newline at end of file
+    def test_no_data(self):
+        blk = TCPStreamer()
+        self.configure_block(blk, {'host': '1.2.3.4', 'port': 12345})
+        with patch('socket.socket') as mock_socket:
+            mock_connect = MagicMock()
+            mock_connect.recv.return_value = None
+            mock_socket.return_value.__enter__.return_value.\
+                accept.side_effect = [(mock_connect, ('1.2.3.4', 12345)),
+                                      Exception]
+
+            blk.start()
+
+            mock_socket.return_value.__enter__.return_value.\
+                bind.assert_called_once_with(('1.2.3.4', 12345))
+            mock_socket.return_value.__enter__.return_value.\
+                listen.assert_called_once_with(1)
+
+            sleep(1)
+            self.assertEqual(len(blk._connections), 0)
+
+            with self.assertRaises(Exception):
+                blk.stop()
```

Add a comment here. It's unclear why this should raise an exception.